### ISSUE TITLE
[Prater] Proposer index is different than calculated

### ISSUE THREAD

**Issue by jclapis** [prysmaticlabs/prysm#11219]
# 🐞 Bug Report

### Description

When calling the API route `/eth/v1/beacon/states/<state>/validators` using a historical state for `<state>` from a few days ago, I ran into this RPC error:

```
HTTP status 500; response body: '{"message":"Invalid state ID: error while replaying history to slot=3661055: could not process block: could not process block header: proposer index: 155125 is different than calculated: 35920","code":500}'
```

The Beacon Node did not show any errors in its logs when this happened, so all I have is this RPC error message.
I suspect this has something to do with Blinded Blocks?

This error is intermittent; trying it again a few minutes later worked properly.

### Has this worked before in a previous version?

I believe this started with v2.1.4, but it could be related to Bellatrix or something else that happened around this time.

## 🔬 Minimal Reproduction

That's a good question, I haven't been able to repro it reliably unfortunately.

## 🔥 Error

See above

## 🌍  Your Environment

**Operating System:**
`Debian 11 x64`

**What version of Prysm are you running? (Which release)**
`v2.1.4` or more accurately, Docker image `prysmaticlabs/prysm-beacon-chain:HEAD-4e225f-debug`

**Comment by rauljordan** [comment]
Maybe the endpoint is not fork-aware @terencechain or @rkapka, like how Terence just fixed #11274

**Comment by rkapka** [comment]
Closing as non-reproducible. We can reopen if someone else runs into the same issue.
